### ISSUE TITLE
Status?

### ISSUE THREAD

**Issue by just4phil** [nooone/gdx-vr#3]
Hi Daniel!
Just a question: how far have you gotten with this?
Can it be used with libgdx already? Is there something people can help with and contribute?

Bye 
Phil

**Comment by nooone** [comment]
Hello Phil,

It is not ready to be used yet. I did come up with some more or less generic API, but it was and still is totally a work in progress. I also implemented a test project with different backends to try it with the different devices. However up to that point it didn't make much sense to keep developing it without having any real devices. I've ordered thoese devices early (e. g. the Oculus DK2), however it took them more than 6 weeks to be delivered. In the meantime I've started another (non-public) project which kept my attention until today.

However, I'm optimistic that I will find more time until christmas or at least new year to keep working on gdx-vr, get it to a more stable API and implement at least the oculus and google cardboard backends completely, so they could be used in games. Unfortunately, up to that point it doesn't make that much sense yet to contribute.

What's your experience with head mounted displays? Are you willing to contribute to the extension itself, or did you ask because you want to use it in a game (or any other application)? If it's the first option we might have a private conversation on IRC maybe and discuss the development further.

**Comment by just4phil** [comment]
Hi!
i have got a google cardboard and was wondering if anybody developed something for it with libgdx.
do you know of anything?
i am not very experienced, but if i could contribute something i would at least try to...

**Comment by joeyenfield** [comment]
Hey, I've got an oculus and a goggle cardboard. I've been looking for a project to help out on. I'd be looking to also integrate the microsoft kinect with it. I'd love to make a nice VR framework.

**Comment by nooone** [comment]
@joeyenfield Microsoft Kinect would be an interesting addition. I planned only head tracking for this extension, but the API I came up with already has `Body` and `Head`. This could be extended and both could be tracked independent from each other via different devices.

Unfortunately I do not own a Kinect and I think we would need to find a Java wrapper for their SDK, as we can't maintain different wrappers as well.

@joeyenfield Have you been working on some framework before? Your github profile is pretty much completely empty. Are you experienced with libgdx itself?

**Comment by joeyenfield** [comment]
Hey, I havn't worked on any open source project before, I normally just work on personal projects messing around with tech usually building my own frameworks. I've been interested in playing with the oculus/cardboard with the kinect and thought that libgdx would be a good approach. I've played with libgdx for a bit so I know my way around it (mainly 2d tho). 

What i'm basically hoping to achieve is a full VR framework. I've looked at some libraries for Kinect. I'm thinking that a good one would be - http://research.dwi.ufl.edu/ufdw/j4k/ its a good enough library. I haven't seen the src but the lib is easy enough and has some good examples. Also it has a nice demo pre-recorded skeletons so you dont need a kinect to work with it. 

The approach that I would take would be to wrap that framework (or any other) in either a socket server or a rest api so that it could be used with either the oculus and the cardboard apps. 

I see that its been a bit of time since you've done any commits, is there any pending commits before I start messing with it?

**Comment by nooone** [comment]
No, currently there are no pending commits. http://research.dwi.ufl.edu/ufdw/j4k/ seems okay on first sight.

> The approach that I would take would be to wrap that framework (or any other) in either a socket server or a rest api so that it could be used with either the oculus and the cardboard apps.

I'm not sure a REST API or a socket server are really necessary here. There should be a common API that gdx-vr offers and the different backends/implementations should fill that API with data. I went with a polling approach. A user would call something like VirtualReality.update() which will trigger the different implementations to collect their sensor data and return the results to the common gdx-vr API.

**Comment by burf2000** [comment]
So, has this died then?  Can it render 2 screens side by side for iOS  / Android?

**Comment by nooone** [comment]
Yes-ish to both questions.

The hardware I got for testing purposes didn't work all too well with my laptop, which kind of killed my motivation to keep going with this.

However, it was able to render the same scene from two perspectives (the eyes) next to each other. That alone is one of the smallest problems though. Anti-distortion for head mounted displays and tracking via the actual devices is a lot harder and gdx-vr was still at the beginning here.

**Comment by burf2000** [comment]
Arr ok, hmm I wonder if you can link it to Googles Cardboard Api as its now ios and Android?

**Comment by nooone** [comment]
Link what to Google Cardboard?

**Comment by burf2000** [comment]
Somehow get Libgdx to use the GoogleCardboard Api's, would that not help?

**Comment by nooone** [comment]
That was actually the plan all along. See the [here](https://github.com/nooone/gdx-vr/tree/master/gdx-vr-android/libs).

**Comment by yangweigbh** [comment]
hi, guys, i have adapted libgdx onto Cardboad sdk, now you can use libgdx to develop cardboad app, welcome to contribute https://github.com/yangweigbh/Libgdx-CardBoard-Extension

**Comment by burf2000** [comment]
Sounds exciting, did you manage to get the iOS Cardboard API working to

**Comment by snapfractalpop** [comment]
@yangweigbh I noticed that your extension has rendering directly in the android code (instead of core). Was there a reasoning behind deviating from the typical structure of a LibGdx app?

**Comment by xeoshow** [comment]
Looking forward to vr!

2016-01-23 6:24 GMT+08:00 Matt Kevins notifications@github.com:
